src/puppetlabs/puppetdb/query_eng.clj

```diff
@@ -86,8 +63,35 @@
   If the query can't be parsed, a 400 is returned."
   [entity version query paging-options db url-prefix]
   (try
-    (let [parsed-query (json/parse-strict-string query true)]
-      (stream-query-result entity version parsed-query paging-options db url-prefix http/stream-json-response))
+    (let [parsed-query (json/parse-strict-string query true)
+          query-error (promise)
+          count-query (promise)
```

Comment:
I find this count-query promise pretty confusing. Why do we need to get the query in such an indirect manner?